pkg/ibmcloudprovider/volume_provider.go

```diff
@@ -72,10 +73,12 @@ func NewIBMCloudStorageProvider(configPath string, logger *zap.Logger) (*IBMClou
 			logger.Fatal("Unable to create API key getter", local.ZapError(err))
 			return nil, err
 		}
-		err = apiKeyImp.UpdateIAMKeys(conf)
-		if err != nil {
-			logger.Fatal("Unable to get API key", local.ZapError(err))
-			return nil, err
+		if os.Getenv("IKS_ENABLED") == "True" {
```

Comment:
Update IAM API key will always be required 

for calling sidecar will be based on the `IKS_ENABLED` true or false

Reply:
ignore this as of now